dask/dataframe/io.py

```diff
@@ -241,12 +243,28 @@ def from_array(x, chunksize=50000):
     name = next(from_array_names)
     dsk = dict(((name, i), (pd.DataFrame,
                              (getitem, x,
-                             (slice(i * chunksize, (i + 1) * chunksize),))))
-            for i in range(0, int(ceil(float(len(x)) / chunksize))))
+                              slice(i * chunksize, (i + 1) * chunksize))))
+            for i in range(0, int(ceil(len(x) / chunksize))))
 
     return DataFrame(dsk, name, columns, divisions)
 
 
+from_dataframe_names = ('from-dataframe-%d' % i for i in count(1))
```

Comment:
In other collections I've switched to having one generator of `tokens`

    tokens = ('-%d' % i for i in count(1))

and then adding names to these

    name = 'from_dataframe' + next(tokens)